style: make border default gray for professional and plh_kids_kw themes

jfmcquade opened the pull request:
PR Checklist

- [x] PR title descriptive (can be used in release notes)

## Description

- `professional` theme: Makes the default border colour grey/gray
- `plh_kids_kw` theme: Makes the default border colour grey/gray
- Minor refactor to default border styles: remove unused global style variables

## Review notes

There should be no changes to component styles for the `default` theme. Below is a comprehensive list of all the components that make use of the variable that has been updated by this PR. If you come across any components that you would expect to be changed in accordance with #2426 but are not affected by this PR, then they do not use the default variable and can be refactored as part of this PR or as a follow-up.

@FaithDaka I've applied these changes to the `plh_kids_kw` theme (see #2426 for justification). I imagine that any custom styling you've been applying as part of that theme or otherwise will override the changes I've made as part of this PR. But it's worth being aware of these changes and identifying whether you'll need to make any updates to your custom styles.

## Git Issues

Closes #2426

## Screenshots/Videos

Below is a comprehensive list of all affected components

| Component Name    | Before | After |
|-------------------|--------|-------|
| audio             | <img width="300" alt="Screenshot 2024-11-04 at 16 58 30" src="https://github.com/user-attachments/assets/8d531376-8f67-40a6-9cb7-b527605edebc">      | <img width="300" alt="Screenshot 2024-11-05 at 12 19 18" src="https://github.com/user-attachments/assets/a5d22559-5fa9-44a5-a252-871c0ab8224f">    |
| accordion         |  <img width="300" alt="Screenshot 2024-11-04 at 16 59 17" src="https://github.com/user-attachments/assets/740b0b35-8d07-42d1-ac3e-2c22e678beba">     | <img width="300" alt="Screenshot 2024-11-05 at 12 20 21" src="https://github.com/user-attachments/assets/a2c5665f-c998-4045-b99b-dbb3463dc2c8">    |
| combo_box         | <img width="300" alt="Screenshot 2024-11-04 at 16 59 44" src="https://github.com/user-attachments/assets/c69006f9-7262-4b09-8dd9-43e723858e70">      | <img width="300" alt="Screenshot 2024-11-05 at 12 35 27" src="https://github.com/user-attachments/assets/9d3d635e-cc6a-4745-b139-a378565cf735">   |
| combo_box_modal   | <img width="300" alt="Screenshot 2024-11-04 at 17 00 06" src="https://github.com/user-attachments/assets/9ad33e06-e3f0-4f91-8a29-c955b0ef0c6a">      | <img width="300" alt="Screenshot 2024-11-05 at 12 35 53" src="https://github.com/user-attachments/assets/017b9400-b501-46db-a020-a7f022a99ac8">    |
| display_grid      | <img width="300" alt="Screenshot 2024-11-04 at 17 00 31" src="https://github.com/user-attachments/assets/a45abd54-da6b-448d-9531-f10b768b9e02">      | <img width="300" alt="Screenshot 2024-11-05 at 12 36 48" src="https://github.com/user-attachments/assets/c90ad828-2862-4a71-b1d8-a9c7f35641a4">    |
| number_selector   | <img width="300" alt="Screenshot 2024-11-04 at 17 00 49" src="https://github.com/user-attachments/assets/e5f8317d-e7fe-464a-8af4-886bbb840257">      | <img width="300" alt="Screenshot 2024-11-05 at 12 37 25" src="https://github.com/user-attachments/assets/6215fa3e-7bcb-4670-9010-7a65c2c58cdf">   |
| radio_button_grid | <img width="300" alt="Screenshot 2024-11-04 at 17 01 08" src="https://github.com/user-attachments/assets/bd2df6d5-87bf-4a05-a306-b8c602ebf616">      | <img width="300" alt="Screenshot 2024-11-05 at 12 38 01" src="https://github.com/user-attachments/assets/f9503636-1a18-46e4-b0ef-0950a68ad87f">   |
| radio_group       | <img width="300" alt="Screenshot 2024-11-04 at 17 01 32" src="https://github.com/user-attachments/assets/8c8beae3-55f9-471d-bd29-310a96243749">      | <img width="300" alt="Screenshot 2024-11-05 at 12 38 29" src="https://github.com/user-attachments/assets/1cbb9ffc-2459-4eca-933c-6b6d8818ec77">    |
| slider            | <img width="300" alt="Screenshot 2024-11-04 at 17 01 48" src="https://github.com/user-attachments/assets/55220db7-1e87-4ce9-a4b2-a03ee5b594c9">      | <img width="300" alt="Screenshot 2024-11-05 at 12 39 02" src="https://github.com/user-attachments/assets/6f10fb5e-f9c4-4430-bcee-3c297c4e1a51">    |
| text_area         | <img width="300" alt="Screenshot 2024-11-04 at 17 02 09" src="https://github.com/user-attachments/assets/d136c865-d3b2-4da4-9053-a68f72057f22">      |  <img width="300" alt="Screenshot 2024-11-05 at 12 39 33" src="https://github.com/user-attachments/assets/10018eae-9dae-462c-b8bd-c3828ffd26cc">    |
| text_box          | <img width="300" alt="Screenshot 2024-11-04 at 17 02 27" src="https://github.com/user-attachments/assets/b6983d00-6cc7-49fe-93b2-043e71b7f88c">      | <img width="300" alt="Screenshot 2024-11-05 at 12 40 05" src="https://github.com/user-attachments/assets/fab9897f-eed9-4bb2-913d-3dc26a9df75c">     |
| timer             | <img width="300" alt="Screenshot 2024-11-04 at 17 02 46" src="https://github.com/user-attachments/assets/1b6c69eb-9030-442e-a069-0dd9ce3736dd">      | <img width="300" alt="Screenshot 2024-11-05 at 12 40 32" src="https://github.com/user-attachments/assets/d0cb6e79-c7b4-496f-b7ee-0321c17617fd">   |

chrismclarke reviewed (commented):
Am I right in thinking that most ionic-based components use the `ion-border-standard` variable, and so this PR mostly exposes that variable to the override system using an author-facing `--border-color-default` variable?

If so the code changes make sense to me, I guess there's the issue of any custom components that have border styles defined standalone, but with or without this PR those would be inconsistent so I think the change should overall be good. 

Although from a design-perspective I don't like the gray. It's the same colour as the drop shadow which breaks the illusion of raised components, leaving just slightly odd sized borders. I'm not sure if this decision is coming direct from design or not, but in any case I'd encourage the default value to have better contrast. See latest material design specs on elevation:
https://m3.material.io/styles/elevation/overview

jfmcquade commented:
> Am I right in thinking that most ionic-based components use the `ion-border-standard` variable, and so this PR mostly exposes that variable to the override system using an author-facing `--border-color-default` variable?
> 
> If so the code changes make sense to me, I guess there's the issue of any custom components that have border styles defined standalone, but with or without this PR those would be inconsistent so I think the change should overall be good.

That was my assumption, although having investigated, ionic doesn't actually use the variable at all. So I've renamed the theme variable(s) `--ion-border-*` -> `--border-*`. There's still a lot of inconsistency within the variable names (using the `ion-` prefix being one example, as this isn't reserved for variables that are consumed by ionic components), but I'm happy not to tidy that up as part of this PR. We never officially moved non-colour variables into the theming system (as you say, there's inconsistency in how custom components apply non-colour variables, and the variables in `src/theme/variables.scss` should be moved over to the theme level and standardised), so this PR makes a start at doing so. 

> Although from a design-perspective I don't like the gray. It's the same colour as the drop shadow which breaks the illusion of raised components, leaving just slightly odd sized borders. I'm not sure if this decision is coming direct from design or not, but in any case I'd encourage the default value to have better contrast. See latest material design specs on elevation: https://m3.material.io/styles/elevation/overview

| M2 | M3 |
|---------|
| <img width="300" alt="lwykc8yv-2" src="https://github.com/user-attachments/assets/fb72d374-f892-4756-b28c-5d5dc2b24b0d"> | <img width="300" alt="lwykckey-3" src="https://github.com/user-attachments/assets/66c1028b-2869-4b4e-a1ec-4ca12ba315ef"> |

I agree, the gray border doesn't look very good. I've updated the border to be 1px rather than 2px and I think this makes a big difference (see updated screenshots in PR description). Now the border feels more like part of the drop shadow, and hopefully reinforces the illusion of raised components rather than breaking it.

The design request as I understand it was to "remove the border" (#2426), so I think they were imagining something broadly inline with the M2 specification. As we don't consistently apply drop-shadows, I think making the border a thin grey is the closest we can get with minimal changes. I do know what you mean though. Ultimately we still have a lot of inconsistency, but I do think the changes included here are an improvement visually.

The changes that @FaithDaka is making for the `plh_kids_kw` deployment feel more in line with the M3 specification in the docs you've linked to, and the plan is to eventually roll those styling changes out across other deployments (at least PLH ones), so I think that should definitely move us in the right direction.

esmeetewinkel commented on src/theme/themes/professional.scss:
@jfmcquade actually I would like to leave the `professional` theme as it is. It's used in Kids TZ and Teens TZ (where this change has not been requested)

chrismclarke commented:
> That was my assumption, although having investigated, ionic doesn't actually use the variable at all. So I've renamed the theme variable(s) `--ion-border-*` -> `--border-*`. There's still a lot of inconsistency within the variable names (using the `ion-` prefix being one example, as this isn't reserved for variables that are consumed by ionic components), but I'm happy not to tidy that up as part of this PR. We never officially moved non-colour variables into the theming system (as you say, there's inconsistency in how custom components apply non-colour variables, and the variables in `src/theme/variables.scss` should be moved over to the theme level and standardised), so this PR makes a start at doing so.

Ah, that makes a lot of sense (not from the code, but as a 'why things are how they are'). Yes I vaguely remember a time when developers would just use `ion` as a generic prefix for variables that they created, which is obviously a huge source of confusion. Glad you were able to identify this and continue to move things in a clearer direction.

> I agree, the gray border doesn't look very good. I've updated the border to be 1px rather than 2px and I think this makes a big difference (see updated screenshots in PR description). Now the border feels more like part of the drop shadow, and hopefully reinforces the illusion of raised components rather than breaking it.

Thanks for revisiting this, yes I think that small change actually makes a huge difference. 

> The design request as I understand it was to "remove the border" (#2426), so I think they were imagining something broadly inline with the M2 specification. As we don't consistently apply drop-shadows, I think making the border a thin grey is the closest we can get with minimal changes. I find it difficult to keep track of the latest design mock-ups, but looking at [this Figma](https://www.figma.com/design/CYnvCNWHbgMdXlqBeBnc4T/ParentApp-for-Kids?node-id=428-4559&node-type=frame&t=ijafah11RdOr227K-0), specifically the "Library Page Mockup - 1" frame, there is a thin grey border around the elements, coupled with a drop-shadow to achieve the desired effect. Likewise some examples from [this figma](https://www.figma.com/design/inDybEiRljIM4KRLyFhIxR/Laurie's-Playground?node-id=0-1&node-type=canvas&t=YJfdtJ2ZIm3WLHk6-0):

Yeah I'm personally not a huge fan of large drop shadows on rounded components anyway (feels a bit outdated, similar to a lot of the 2009/10 designs from https://www.toptal.com/designers/ui/button-design-dribbble-timeline). I think ionic themselves also started to move away from this style a few years back but our design defaults typically fallback to the early design drafts that were more like these. So nice if our defaults do start to better-reflect more common modern design practices.

> ![1](https://private-user-images.githubusercontent.com/64838927/383496210-9ba0d24c-5a62-4afd-b0b2-b81457b0b015.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5MzE3ODksIm5iZiI6MTczMDkzMTQ4OSwicGF0aCI6Ii82NDgzODkyNy8zODM0OTYyMTAtOWJhMGQyNGMtNWE2Mi00YWZkLWIwYjItYjgxNDU3YjBiMDE1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA2VDIyMTgwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWExMWM5YjRiOThhYzI0YWE3Mjk5YWU2MGNkZTAxMzA0MDY4YWEyMGI2YzM2N2E5OGQxZDExNTMwZGJjZTkwMDcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.vwCuQ9-_BytE6i1nVWCRTxcEBXovYl7php9lMqOW3Ik) ![2](https://private-user-images.githubusercontent.com/64838927/383496247-3de0cbd6-9fe6-4550-bed7-5be233e2051c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5MzE3ODksIm5iZiI6MTczMDkzMTQ4OSwicGF0aCI6Ii82NDgzODkyNy8zODM0OTYyNDctM2RlMGNiZDYtOWZlNi00NTUwLWJlZDctNWJlMjMzZTIwNTFjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA2VDIyMTgwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThmZDhlYzcxYjJkODRlNDAwYWFkYThkYjkwY2RjNzViM2NlMGVjYmI3NDc0ZGFkMjJkZGFkZGVhZmIzYjU2ZGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.SwlaRivsuI5AVD1K85-6MYNBXbTVp5h1PJzABU4dDlM) ![3](https://private-user-images.githubusercontent.com/64838927/383496273-9a85ec4c-bb96-4ee6-a238-3499445c317a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA5MzE3ODksIm5iZiI6MTczMDkzMTQ4OSwicGF0aCI6Ii82NDgzODkyNy8zODM0OTYyNzMtOWE4NWVjNGMtYmI5Ni00ZWU2LWEyMzgtMzQ5OTQ0NWMzMTdhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA2VDIyMTgwOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwZWRkMjU5OTA3YTBkMDZmZGU5MTdhZTU5YmYwY2NjZTQyYmFmZjEwOGEwYzA2NTRlOTI4YzI4NmM2ODQ4YTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.mJpJXkdzyq_UbvqdLCzl5tj_r87FkNK_zPZ85ceRhp0)

> I do know what you mean though. Ultimately we still have a lot of inconsistency, but I do think the changes included here are an improvement visually.

Agreed, and if we just keep making these minor changes as things get raised then I think over time hopefully most of the codebase should continue to improve without the need for a major overhaul. 

> The changes that @FaithDaka is making for the `plh_kids_kw` deployment feel more in line with the M3 specification in the docs you've linked to, and the plan is to eventually roll those styling changes out across other deployments (at least PLH ones), so I think that should definitely move us in the right direction.
